test/integration/targets/loops/tasks/main.yml

```
@@ -334,3 +334,87 @@
     - 1
   loop_control:
     loop_var: alvin
+
+
+- name: Set handler variables to known value
+  set_fact:
+    handler_one: null
+    handler_two: null
+
+- name: Test loop_control.notify_scope=task
+  shell: "{{ item.cmd }}"
+  notify: "{{ item.notify }}"
+  changed_when: ansible_loop.first
+  loop:
+    - cmd: echo ONE
+      notify:
+        - Handler one
+    - cmd: echo TWO
+      notify:
+        - Handler two
+  loop_control:
+    extended: yes
+
+- meta: flush_handlers
+
+- name: Test all handlers were run
+  assert:
+    that:
+      - handler_one != None
```

Using the test rather than a comparison is a bit more reliable. Types with Jijna can be a pain.
```suggestion
      - handler_one is not none
```